phoenix-core/src/it/java/org/apache/phoenix/end2end/OrphanChildLinkRowsIT.java

```diff
@@ -0,0 +1,201 @@
+            connection.createStatement().execute(v2);
+        }
+        catch (TableAlreadyExistsException e) {
+        }
+
+        try (Connection connection = DriverManager.getConnection(getUrl())) {
+            // wait for TASK to complete
+            waitForTaskState(connection, PTable.TaskType.CHILD_LINK_SCAN, SYSTEM_CHILD_LINK_TABLE, PTable.TaskStatus.COMPLETED);
+
+            // scan the physical table and check there are no UNVERIFIED rows
+            PTable childLinkPTable = PhoenixRuntime.getTable(connection, SYSTEM_CHILD_LINK_NAME);
+            byte[] emptyCF = SchemaUtil.getEmptyColumnFamily(childLinkPTable);
+            byte[] emptyCQ = EncodedColumnsUtil.getEmptyKeyValueInfo(childLinkPTable).getFirst();
+            Scan scan = new Scan();
+            HTable table = (HTable) connection.unwrap(PhoenixConnection.class).getQueryServices().getTable(TableName.valueOf(SYSTEM_CHILD_LINK_NAME).getName());
+            ResultScanner results = table.getScanner(scan);
+            Result result = results.next();
+            while (result != null) {
+                Assert.assertTrue("Found Child Link row with UNVERIFIED status", Arrays.equals(result.getValue(emptyCF, emptyCQ), VERIFIED_BYTES));
+                result = results.next();
+            }
+        }
+    }
+
+    /**
+     * Do 10 times: Create 2 tables and view with same name on both tables.
+     * Check if LIMIT query on SYSTEM.CHILD_LINK returns the right number of rows
+     */
+    @Test
+    public void testChildLinkQueryWithLimit() throws Exception {
+
+        ConnectionQueryServicesImpl.setFailPhaseThreeChildLinkWriteForTesting(true);
+        ChildLinkScanTask.disableChildLinkScanTask(true);
+
+        String CREATE_TABLE_DDL = "CREATE TABLE %s (TENANT_ID VARCHAR NOT NULL, A INTEGER NOT NULL, B INTEGER CONSTRAINT PK PRIMARY KEY (TENANT_ID, A))";
+        String CREATE_VIEW_DDL = "CREATE VIEW %s (NEW_COL1 INTEGER, NEW_COL2 INTEGER) AS SELECT * FROM %s WHERE B > 10";
+
+        try (Connection connection = DriverManager.getConnection(getUrl())) {
+            for (int i=0; i<10; i++) {
+                String table1 = "T_" + generateUniqueName();
+                String table2 = "T_" + generateUniqueName();
+                String view = "V_" + generateUniqueName();
+                connection.createStatement().execute(String.format(CREATE_TABLE_DDL, table1));
+                connection.createStatement().execute(String.format(CREATE_TABLE_DDL, table2));
+                connection.createStatement().execute(String.format(CREATE_VIEW_DDL, view, table1));
+                expectedChildLinks.put(table1, view);
+                try {
+                    connection.createStatement().execute(String.format(CREATE_VIEW_DDL, view, table2));
+                }
+                catch (TableAlreadyExistsException e) {
+
+                }
+            }
+
+            String childLinkQuery = "SELECT * FROM SYSTEM.CHILD_LINK LIMIT 7";
```

Comment:
nit : why not use SELECT count(*) instead?

Reply:
@jpisaac I wanted to test the LIMIT functionality to make sure the scanner implementation is correct, as per a previous comment from Kadir.